checks/raw/branch_protection.go

```diff
@@ -51,7 +51,8 @@ func (set branchSet) contains(branch string) bool {
 }
 
 // BranchProtection retrieves the raw data for the Branch-Protection check.
-func BranchProtection(c clients.RepoClient) (checker.BranchProtectionsData, error) {
+func BranchProtection(cr *checker.CheckRequest) (checker.BranchProtectionsData, error) {
+	c := cr.RepoClient
```

Comment:
Change the parameter from `clients.RepoClient` to `*checker.CheckRequest` to align it with similar APIs for other checks.